web/src/portico/help.js

```diff
@@ -123,11 +123,11 @@ function render_code_sections() {
 function scrollToHash(simplebar) {
     const hash = window.location.hash;
     const scrollbar = simplebar.getScrollElement();
-    if (hash !== "" && $(hash).length > 0) {
-        const position = $(hash).position().top - $(scrollbar.firstChild).position().top;
+    const $scroll_target = $(CSS.escape(hash));
```

Comment:
No, this is still wrong. We need to

1. remove the initial `#`, which represented the separation between the URL path and the fragment identifier; then
2. `CSS.escape` *only the part after that*, which was the actual ID of the DOM node the fragment refers to; then
3. stick an *unescaped* `#` onto the front, to introduce a CSS selector by ID.

This is necessary to guarantee that the result is a single well-formed CSS selector that’s safe to pass to jQuery.

```js
» $(CSS.escape("#related-articles"))  // wrong
Object { length: 0, prevObject: {…} }

» $(`#${CSS.escape("related-articles")}`)  // right
Object { 0: h2#related-articles.scroll-target, length: 1 }
```

Reply:
I’ve removed this code completely in #15713.